Models/USBSpectrometer.cs

```diff
@@ -0,0 +1,696 @@
+            pixels = eeprom.activePixelsHoriz;
+
+            await updateBatteryAsync();
+
+            // for now, ignore EEPROM configuration and hardcode
+            // integrationTimeMS = (ushort)(eeprom.startupIntegrationTimeMS > 0 && eeprom.startupIntegrationTimeMS < 5000 ? eeprom.startupIntegrationTimeMS : 400);
+            // gainDb = eeprom.detectorGain;
+            integrationTimeMS = 400;
+            gainDb = 8;
+
+            verticalROIStartLine = eeprom.ROIVertRegionStart[0];
+            verticalROIStopLine = eeprom.ROIVertRegionEnd[0];
+
+            logger.info($"initialized {eeprom.serialNumber} {fullModelName}");
+            logger.info($"  detector: {eeprom.detectorName}");
+            logger.info($"  pixels: {pixels}");
+            logger.info($"  integrationTimeMS: {integrationTimeMS}");
+            logger.info($"  gainDb: {gainDb}");
+            // logger.info($"  verticalROI: ({verticalROIStartLine}, {verticalROIStopLine})");
+            logger.info($"  excitation: {laserExcitationNM:f3}nm");
+            logger.info($"  wavelengths: ({wavelengths[0]:f2}, {wavelengths[pixels - 1]:f2})");
+            if (wavenumbers != null)
+                logger.info($"  wavenumbers: ({wavenumbers[0]:f2}, {wavenumbers[pixels - 1]:f2})");
+
+            // I'm honestly not sure where we should initialize location, but it 
+            // should probably happen after we've successfully connected to a
+            // spectrometer and are ready to take measurements.  Significantly,
+            // at this point we know the user has already granted location privs.
+            //
+            // whereAmI = WhereAmI.getInstance();
+
+            logger.debug("Spectrometer.initAsync: done");
+            return true;
+        }
+        public void connect()
+        {
+            if (measurement is null)
+                measurement = new Measurement();
+
+            if (!isConnected)
+            {
+                bool ok = udc.SetConfiguration(acc.GetConfiguration(0));
+                if (ok)
+                {
+                    logger.info("successfully set configuration");
+                    ok = udc.ClaimInterface(acc.GetInterface(0), false);
+                    if (ok)
+                    {
+                        logger.info("successfully claimed interface");
+                        isConnected = true;
+                    }
+                }
+            }
+        }
+
+        public override void reset()
+        {
+            logger.debug("Spectrometer.reset: start");
+            paired = false;
+
+            // Provide some test defaults so we can play with the chart etc while
+            // disconnected.  These will all be overwritten when we read an EEPROM.
+            pixels = 1952;
+            laserExcitationNM = 785.0f;
+            wavelengths = new double[pixels];
+            for (int i = 0; i < pixels; i++)
+                wavelengths[i] = laserExcitationNM + 15 + i / 10.0;
```

Comment:
Whut

Reply:
I copied this from your code! I think it's just so the chart doesn't vomit if there's a non calibrated spec connected